vmp/resources/waf/custom_rule_set.go

```diff
@@ -0,0 +1,615 @@
+// Copyright Verizon Media, Licensed under the terms of the Apache 2.0 license . See LICENSE file in project root for terms.
+
+package waf
+
+import (
+	"context"
+	"errors"
+	"log"
+	"terraform-provider-vmp/vmp/api"
+	"terraform-provider-vmp/vmp/helper"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	sdkwaf "github.com/EdgeCast/ec-sdk-go/edgecast/waf"
+)
+
+func ResourceCustomRuleSet() *schema.Resource {
+
+	return &schema.Resource{
+		CreateContext: ResourceCustomRuleSetCreate,
+		ReadContext:   ResourceCustomRuleSetRead,
+		UpdateContext: ResourceCustomRuleSetUpdate,
+		DeleteContext: ResourceCustomRuleSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"account_number": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Identifies your account by its customer account number.",
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				Description:  "Indicates the name of the custom rule.",
+			},
+			"directive": {
```

Comment:
Not at all required to merge this, but I wonder if it is possible to reduce the indentations to come closer to our 80 character wide target. It might make this unreadable, though, since this is pretty in depth nesting. A good topic for our tech discussions.

Reply:
I left the resource indentations as is, since that would render this unreadable. I have however updated the indentations of the other functions below. Thanks.